pkg/primitives/public_ip.go

```diff
@@ -45,9 +47,10 @@ func (p *Primitives) getAssignedPublicIP(ctx context.Context, wl *gridtypes.Work
 	reserved := contract.PublicIPs
 
 	// make sure we have enough reserved IPs
-	ipWorkloads := deployment.ByType(zos.PublicIPType)
+	// we process both ipv4 type and ip type to be backward compatible
+	ipWorkloads := deployment.ByType(zos.PublicIPv4Type, zos.PublicIPType)
 	if len(ipWorkloads) > len(reserved) {
```

Comment:
ipWorkloads can contain ipv6-only workloads which doesn't have to have a corresponding reserved ip.

Reply:
True, good catch